articles/preparing_for_consoles.md

```diff
@@ -0,0 +1,112 @@
+
+Reflection is, at large, unsupported in an AOT context. Which means that anywhere you or a third party library uses reflection, your game will crash.
+
+In order to make sure that your game abides to that rule, you can try to publish an AOT'd version of it for desktop computers and verify how it fares.
+
+To publish with AOT:
+
+- Add `<PublishAot>true</PublishAot>` to your `.csproj`.
+- Then run `dotnet publish` from the command-line/terminal.
+
+This will nativily compile your game in a fashion similar to consoles. The output executable will be in your output folder, within a sub-folder nammed `publish`.
+
+From there, you can try running this executable. If it does not start or crashes later on, you likely are using reflection or another unsupported feature in a AOT runtime.
+
+Native executables can debugged by:
+
+- Atarting an empty Visual Studio with no code.
+- Opening the compiled exe.
+- Then hitting "Start debugging".
+
+It should show you on which C# lines it crashed.
+
+It is important to note that you should test every aspect of your game, and not just if it starts.  Run through all the menus/screens, scene transitions and gameplay.
+
+Another way to make sure that everything is safe is to enable `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` in your `.csproj`, and then **rebuild** (not just build) your game and check the build output for AOT warnings. Those warnings will tell you which parts of your code might trigger crashes or unexecpted results when running on AOT compilation. You should seek to resolve all of them.
+
+## No runtime compilation / IL emit
+
+Generating code at runtime is a scenario that is also not supported in AOT contexts, it is also forbidden by console manufacturers.
+
+Like reflection, trying to make a `PublishAot` build if a good way to verify that your game is compliant because any use of IL emit will crash.
+
+## No use of dynamic assembly loading
+
+Loading assemblies at runtime with `Assembly.Load()` is not supported.
+
+## Third party libraries
+
+Many third party libraries heavily rely on using reflection or IL emit, this is a common practice for JSON or XML parsers for example.
+
+It is advised to choose very carefully the libraries that you are using when porting to consoles. If you do not select them with this in mind, you might run into a situation in which you will have to rewrite entire chunks of data handling.
+
+The best way to make sure if they will work, is to search if they are **"AOT-compatible"**, or try to compile with the `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` setting in your `.csproj` and check if there are any warnings related to those libraries.
+
+## Native libraries
+
+If you are using native libraries, make sure that they propose builds for consoles, or make sure that you can compile and run them on consoles yourself.
```

Comment:
`make sure that they propose builds for consoles` should be `make sure that they provide builds for consoles` ?

Reply:
Updated